server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DBStructureComparisonFlowableTask.java

```diff
@@ -15,50 +15,127 @@
+import com.oceanbase.odc.core.shared.constant.TaskStatus;
+import com.oceanbase.odc.metadb.task.TaskEntity;
+import com.oceanbase.odc.service.flow.task.model.DBStructureComparisonParameter;
+import com.oceanbase.odc.service.flow.util.FlowTaskUtil;
+import com.oceanbase.odc.service.structurecompare.StructureComparisonContext;
+import com.oceanbase.odc.service.structurecompare.StructureComparisonService;
 import com.oceanbase.odc.service.task.TaskService;
 
+import lombok.extern.slf4j.Slf4j;
+
 /**
  * @author jingtian
  * @date 2024/1/9
- * @since
+ * @since ODC_release_4.2.4
  */
-public class DBStructureComparisonFlowableTask extends BaseODCFlowTaskDelegate<DBStructureComparisonTaskResult> {
+@Slf4j
+public class DBStructureComparisonFlowableTask extends BaseODCFlowTaskDelegate<Void> {
+    private volatile StructureComparisonContext context;
+    @Autowired
+    private StructureComparisonService service;
+
     @Override
-    protected DBStructureComparisonTaskResult start(Long taskId, TaskService taskService, DelegateExecution execution)
+    protected Void start(Long taskId, TaskService taskService, DelegateExecution execution)
             throws Exception {
+        TaskEntity taskEntity = taskService.detail(taskId);
+        if (taskEntity == null) {
+            throw new IllegalStateException("Can not find task entity by id " + taskId);
+        }
+        log.info("Structure comparison task starts, taskId={}, activityId={}", taskId,
+                execution.getCurrentActivityId());
+
+        DBStructureComparisonParameter parameters = FlowTaskUtil.getDBStructureComparisonParameter(execution);
+        Validate.notNull(parameters, "Structure comparison task parameters can not be null");
+        Long flowInstanceId = FlowTaskUtil.getFlowInstanceId(execution);
+
+        context = service.create(parameters, taskId, taskEntity.getCreatorId(), flowInstanceId);
+        taskService.start(taskId, context.getStatus());
         return null;
     }
 
+    @Override
+    protected boolean cancel(boolean mayInterruptIfRunning, Long taskId, TaskService taskService) {
+        if (context == null) {
+            throw new IllegalStateException("Context is null, structure comparison task may not be running");
+        }
+        boolean result = context.cancel(true);
+        log.info("structure comparison task has been cancelled, taskId={}, result={}", taskId, result);
+        taskService.cancel(taskId, context.getStatus());
+        return result;
+    }
+
+    @Override
+    public boolean isCancelled() {
+        if (context == null) {
+            return false;
+        }
+        return context.isCancelled();
+    }
+
     @Override
     protected boolean isSuccessful() {
-        return false;
+        if (context == null || !context.isDone()) {
+            return false;
+        }
+        try {
+            return context.get(1, TimeUnit.MILLISECONDS).getStatus() == TaskStatus.DONE;
```

Comment:
can not understand this, why is it different between `isSuccessful` and `isFailure`?